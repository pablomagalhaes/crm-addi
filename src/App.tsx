import React, { FC } from 'react';
import { Provider } from 'react-redux';
import { hot } from 'react-hot-loader';
import { ThemeProvider } from 'styled-components';
import { PersistGate } from 'redux-persist/integration/react';

import { persistor, store } from './io/redux/store';
import theme from './theme';
import Routes from './Routes';

const App: FC = () => (
  <Provider store={store}>
    <PersistGate loading={null} persistor={persistor}>
      <ThemeProvider theme={theme}>
        <Routes />
      </ThemeProvider>
    </PersistGate>
  </Provider>
);

export default hot(module)(App);
