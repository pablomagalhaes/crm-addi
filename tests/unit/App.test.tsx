import React from 'react';
import { shallow } from 'enzyme';

import App from '@/App';

describe('<App />', () => {
  it('should match snapchat', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
