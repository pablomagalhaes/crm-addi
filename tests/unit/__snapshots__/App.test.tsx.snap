// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapchat 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <PersistGate
    loading={null}
    persistor={
      Object {
        "dispatch": [Function],
        "flush": [Function],
        "getState": [Function],
        "pause": [Function],
        "persist": [Function],
        "purge": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Component
      theme={
        Object {
          "borderRadius": Object {
            "default": "3px",
          },
          "breakpoint": Object {
            "lg": 992,
            "md": 768,
            "sm": 576,
            "xl": 1300,
          },
          "color": Object {
            "background": "#ebebeb",
            "black": "#424242",
            "border": "rgba(0,0,0,0.1)",
            "danger": "#FF5B68",
            "primary": "#6272a4",
            "secondary": "#44475a",
            "success": "#2AD178",
            "warning": "#FABD59",
            "white": "#FFF",
          },
          "font": Object {
            "color": Object {
              "primary": "#424242",
              "secondary": "#969696",
            },
            "family": Object {
              "primary": "'Open Sans', sans-serif",
            },
            "weight": Object {
              "regular": 400,
              "semiBold": 600,
            },
          },
        }
      }
    >
      <Routes />
    </Component>
  </PersistGate>
</Provider>
`;
